geode-core/src/main/java/org/apache/geode/internal/cache/DiskStoreFactoryImpl.java

```diff
@@ -141,10 +141,9 @@ DiskStoreImpl createOwnedByRegion(String name, boolean isOwnedByPR,
   @Override
   public DiskStore create(String name) {
     this.attrs.name = name;
-    // As a simple fix for 41290, only allow one DiskStore to be created
-    // at a time per cache by syncing on the cache.
     DiskStore result;
-    synchronized (this.cache) {
+    try {
```

Comment:
Based on the previous comment for 41290; it seems there was a need to create disk-store one at a time. 
From the comment its not clear, if its when the disk-stores are created through java API or cache xml (ClusterConfig).
Why do we need to change the synchronization...Here the disk-stores are just added into the map and reading from the disk happens in the CacheCreation.java.

Reply:
Before I created this pull request, I actually looked into 41290, which was created more than 10 years ago. It was a totally different implementation. 41290 is no longer a concern.

We do need to remove the synchronization, in order to make the disk store recovery parallel. This synchronized block not only adds disk stores into the map, but also reads the disks, please see `initializeDiskStore(dsi);` in line 157. `initializeDiskStore()` eventually invokes `Oplog.recoverCrf()` and `Oplog.readKrf()`. Without removing the synchronization, the disk store can only be recovered one at a time.